Tasks/KubernetesManifestV0/src/actions/deploy.ts

```diff
@@ -0,0 +1,156 @@
+"use strict";
+import tl = require('vsts-task-lib/task');
+import path = require('path');
+import fs = require("fs");
+import * as utils from "./../utilities";
+import Q = require('q');
+import { IExecSyncResult } from 'vsts-task-lib/toolrunner';
+import kubectlutility = require("utility-common/kubectlutility");
+import { Kubectl } from "utility-common/kubectl-object-model";
+
+var kubectlPath = "";
+var execResults: IExecSyncResult[] = [];
+
+export async function deploy() {
+    execResults = [];
+    
+    var files: string[] = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory") || process.cwd(), tl.getInput("manifests", true));
+    
+    if (files.length == 0) {
+        throw ("No file found with matching pattern");
+    }
+
+    let containers = tl.getDelimitedInput("containers", "\n");
+    files = updateContainerImagesInConfigFiles(files, containers);
+
+    let kubectl = new Kubectl(await getKubectl(), tl.getInput("namespace", false));
+
+    for (let i = 0; i < files.length; i++) {
+        let filePath = files[i];
+        let results = kubectl.apply(filePath);
```

Comment:
Instead of calling apply for each file separately, call it once with all files together